File: PyProject/leetcode/link_table/fan-zhuan-lian-biao-lcof.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


#     ListNode reverse = reverseList(head.next);
#     head.next.next = head;
#     head.next = null;
#     return reverse;
#

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if not head or head.next:
            return head
        after = head.next
        cur = self.reverseList(after)
        cur.next = head
        head.next = None
        return cur

File: GoProject/leetcode/tenxun/linkedList/fan-zhuan-lian-biao-lcof.go
package linkedList

type ListNode struct {
	Val  int
	Next *ListNode
}

func helperReverseList(head *ListNode) (*ListNode, *ListNode) {
	if head == nil || head.Next == nil {
		return head, head
	}
	node, root := helperReverseList(head.Next)
	node.Next = head
	head.Next = nil
	return head, root

}

func reverseList(head *ListNode) *ListNode {
	_, r := helperReverseList(head)
	return r
}
